apps/prairielearn/src/sync/course-db.ts

```diff
@@ -239,10 +240,24 @@ export async function loadFullCourse(
     // expired if it either has zero `allowAccess` rules (in which case it is never
     // accessible), or if it has one or more `allowAccess` rules and they all have
     // an `endDate` that is in the past.
-    const allowAccessRules = courseInstance.data?.allowAccess ?? [];
-    const courseInstanceExpired = allowAccessRules.every((rule) => {
-      const endDate = rule.endDate ? parseJsonDate(rule.endDate) : null;
-      return endDate && isPast(endDate);
+
+    // If the allowAccess section is not present, we instead consider publishing.endDate.
```

Comment:
Nit, to make it clear that this is describing a block and not the following line:

```suggestion
    //
    // If the allowAccess section is not present, we instead consider publishing.endDate.

```